test/unit/gateways/cyber_source_test.rb

```diff
@@ -877,6 +877,69 @@ def test_adds_3ds2_fields_via_normalized_hash
     end.respond_with(successful_purchase_response)
   end
 
+  def test_does_not_add_3ds2_fields_via_normalized_hash_when_cavv_and_commerce_indicator_absent
+    options = options_with_normalized_3ds(cavv: nil, commerce_indicator: nil)
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, options)
+    end.check_request do |_, data, _|
+      assert_not_match(/<eciRaw\>#{options[:three_d_secure][:eci]}</, data)
+      assert_not_match(/<cavv\>#{options[:three_d_secure][:cavv]}</, data)
+      assert_not_match(/<paSpecificationVersion\>#{options[:three_d_secure][:version]}</, data)
+      assert_not_match(/<directoryServerTransactionID\>#{options[:three_d_secure][:ds_transaction_id]}</, data)
+      assert_not_match(/<paresStatus\>#{options[:three_d_secure][:authentication_response_status]}</, data)
+      assert_not_match(/<cavvAlgorithm\>#{options[:three_d_secure][:cavv_algorithm]}</, data)
+      assert_not_match(/<veresEnrolled\>#{options[:three_d_secure][:enrolled]}</, data)
+      assert_not_match(/<commerceIndicator\>#{options[:commerce_indicator]}</, data)
+    end.respond_with(successful_purchase_response)
+  end
+
+  def test_adds_3ds2_fields_via_normalized_hash_when_cavv_and_commerce_indicator_absent_and_commerce_indicator_not_inferred
+    @credit_card.brand =
+      (ActiveMerchant::Billing::CyberSourceGateway.supported_cardtypes -
+        ActiveMerchant::Billing::CyberSourceGateway::ECI_BRAND_MAPPING.keys).first
```

Comment:
Hello Bertrand! Could you add some context on why we need to do this? When I debugged locally the card brand that was returned was `diners_club`, is this expected?

Reply:
done

Reply:
done